pvlib/modelchain.py

```diff
@@ -349,6 +369,259 @@ def __init__(self, system, location,
         self.times = None
         self.solar_position = None
 
+    @classmethod
+    def with_pvwatts(cls, system, location,
+                     orientation_strategy=None,
+                     clearsky_model='ineichen',
+                     airmass_model='kastenyoung1989',
+                     temperature_model=None,
+                     name=None,
+                     **kwargs):
+        """
+        ModelChain that follows the PVWatts methods.
+
+        Parameters
+        ----------
+        system : PVSystem
+            A :py:class:`~pvlib.pvsystem.PVSystem` object that represents
+            the connected set of modules, inverters, etc.
+
+        location : Location
+            A :py:class:`~pvlib.location.Location` object that represents
+            the physical location at which to evaluate the model.
+
+        orientation_strategy : None or str, default None
+            The strategy for aligning the modules. If not None, sets the
+            ``surface_azimuth`` and ``surface_tilt`` properties of the
+            ``system``. Allowed strategies include 'flat',
+            'south_at_latitude_tilt'. Ignored for SingleAxisTracker systems.
+
+        clearsky_model : str, default 'ineichen'
+            Passed to location.get_clearsky.
+
+        airmass_model : str, default 'kastenyoung1989'
+            Passed to location.get_airmass.
+
+        temperature_model: None, str or function, default None
+            Valid strings are 'sapm', 'pvsyst', and 'faiman'. The ModelChain
+            instance will be passed as the first argument to a user-defined
+            function.
+
+        name: None or str, default None
+            Name of ModelChain instance.
+
+        **kwargs
+            Arbitrary keyword arguments. Included for compatibility, but not
+            used.
+
+        Examples
+        --------
+        >>> module_parameters = dict(gamma_pdc=-0.003, pdc0=1)
+        >>> inverter_parameters = dict(pac0=1)
+        >>> system = PVSystem(surface_tilt=30, surface_azimuth=180,
+        ...     module_parameters=module_parameters,
+        ...     inverter_parameters=inverter_parameters)
+        >>> location = Location(32.2, -110.9)
+        >>> ModelChain.with_pvwatts(system, location)
+        ModelChain:
+          name: None
+          orientation_strategy: None
+          clearsky_model: ineichen
+          transposition_model: perez
+          solar_position_method: nrel_numpy
+          airmass_model: kastenyoung1989
+          dc_model: pvwatts_dc
+          ac_model: pvwatts_inverter
+          aoi_model: physical_aoi_loss
+          spectral_model: no_spectral_loss
+          temperature_model: sapm_temp
+          losses_model: pvwatts_losses
+        """
+
+        kwargs.update(PVWATTS_CONFIG)
```

Comment:
If there is overlap in kwargs, this prioritizes `PVWATTS_CONFIG`, e.g. if the user passes `transposition_model='isotropic'`, `'perez'` will get used anyway. Should these lines should be `config = PVWATTS_CONFIG.copy(); config.update(kwargs)` so that user-supplied parameters take precedence over the defaults?  Or maybe if people want to mix and match models, they should just use the main ModelChain constructor.

Reply:
Thanks. Yes and yes. I'll change the code, document the behavior, and encourage use of the main constructor for maximum flexibility.